src/main/java/at/hannibal2/skyhanni/features/misc/GeorgeDisplay.kt

```diff
@@ -0,0 +1,132 @@
+import at.hannibal2.skyhanni.events.InventoryOpenEvent
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.cleanName
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzRarity
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.enumJoinToPattern
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.NEUItems.getPriceOrNull
+import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.OSUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.StringUtils.convertToInternalNameString
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class GeorgeDisplay {
+
+    private val config get() = SkyHanniMod.feature.misc.pets.george
+    private val useFandomWiki get() = SkyHanniMod.feature.commands.betterWiki.useFandom
+
+    private val SEPARATOR = ";"
+
+    private val patternGroup = RepoPattern.group("george.tamingcap")
+    private val neededPetPattern by patternGroup.pattern(
+        "needed.pet.loreline",
+        "(?i) +(?<fullThing>(?<tierColorCodes>§.)*(?<tier>${enumJoinToPattern<LorenzRarity>{it.rawName.lowercase()}}) (?<pet>[\\S ]+))"
+    )
+    private val offerPetsChestPattern by patternGroup.pattern(
+        "offerpets.chestname",
+        "Offer Pets"
+    )
+    private val increaseCapItemPattern by patternGroup.pattern(
+        "increasecap.itemname",
+        "\\+1 Taming Level Cap"
+    )
+
+    private var display = listOf<Renderable>()
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryOpenEvent) {
+        if (!isEnabled()) return
+        if (!offerPetsChestPattern.matches(event.inventoryName)) return
+        val stack = event.inventoryItems[41] ?: return //the slot #41 = spawn egg with the tooltip
+        if (!increaseCapItemPattern.matches(stack.cleanName())) return
+        display = drawDisplay(stack.getLore())
+    }
+
+    private fun drawDisplay(lore: List<String>) = buildList<Renderable> {
+        this.add(
+            Renderable.string(
+                "§d§lTaming 60 Cost: §r§d(${
+                    if (config.otherRarities) "cheapest" else "exact"
+                } rarity)"
+            )
+        )
+        var totalCost = 0.0
+        for (line in lore)
+            neededPetPattern.matchMatcher(line) {
+                val origTierString = group("tier")
+                val origPetString = group("pet")
+                val pet = origPetString.convertToInternalNameString().removePrefix("FROST_")
```

Comment:
Please extract the inside of the matchMatcher block into a function to keep indentations shorter